### PR TITLE
imotality todolist 

### PR DESCRIPTION
**Pull request by imotality**

### CODE REVIEW

File: Yurii_Telishevskyi/todo/src/app/todocomponent/todocomponent.component.html [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<div class="design1">
+    <input type="checkbox"  (change)="checked()">
+    <span [class.line]="isGood">{{ items }}</span>
```
Comment (KolosUK) [inline Yurii_Telishevskyi/todo/src/app/todocomponent/todocomponent.component.html]: Re-name your bool property, you need to write the more clear name

File: Yurii_Telishevskyi/todo/src/app/todocomponent/todocomponent.component.html [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+    <button [disabled]="isGood" (click)="isEdited()" class = "m-1 p-1 btn btn-outline-primary"> Edit </button>
```
Comment (KolosUK) [inline Yurii_Telishevskyi/todo/src/app/todocomponent/todocomponent.component.html]: Name for click for example need contains action, for bool property start with "is"
(click) = "edit()"
[disable]="isEditable"

File: Yurii_Telishevskyi/todo/src/app/todocomponent/todocomponent.component.html [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<div class="design1">
```
Comment (KolosUK) [inline Yurii_Telishevskyi/todo/src/app/todocomponent/todocomponent.component.html]: Re-name your class name

File: Yurii_Telishevskyi/todo/src/app/todolists/todolists.component.html [hunk 4]
```diff
@@ -0,0 +1,24 @@
+<div class="d-flex  flex-column h-100 align-items-center justify-content-center">
+  <div class="col-6">
+    <label class="align"> Enter some text </label>
+  <div>
+    <input class="align" type="text" name="todo" [(ngModel)]="todo" placeholder="text">
```
Comment (KolosUK) [inline Yurii_Telishevskyi/todo/src/app/todolists/todolists.component.html]: Add a clear name for your model

File: Yurii_Telishevskyi/todo/src/app/todolists/todolists.component.html [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    <button class="btn btn-primary align" 
+    [disabled]= "!todo" 
+    (click)="addToList()" 
+    type="button">{{btnAdd}}</button>
```
Comment (KolosUK) [inline Yurii_Telishevskyi/todo/src/app/todolists/todolists.component.html]: Re-name "btnAdd"

File: Yurii_Telishevskyi/todo/src/app/todolists/todolists.component.html [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  </div>
+  <div class ="border rounded ">
+    <ul style="list-style-type:none;">
```
Comment (KolosUK) [inline Yurii_Telishevskyi/todo/src/app/todolists/todolists.component.html]: A better solution to add style to .scss file

File: Yurii_Telishevskyi/todo/src/app/todolists/todolists.component.ts [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+  templateUrl: './todolists.component.html',
+  styleUrls: ['./todolists.component.scss']
+})
+export class TodolistsComponent {
+
+  title = 'todo';
+  todo: string;
+  text: string;
+  list: string[] = [];
+  btnAdd = 'Add';
```
Comment (KolosUK) [inline Yurii_Telishevskyi/todo/src/app/todolists/todolists.component.ts]: Create string constant for this

File: Yurii_Telishevskyi/todo/src/app/todolists/todolists.component.ts [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+  i: number;
+  addToList() {
+    if (this.btnAdd === 'Add') {
+      this.list.push(this.todo) ;
+      this.todo = '';
+    } else {
+    this.list[this.i] = this.todo;
+    this.btnAdd = 'Add';
```
Comment (KolosUK) [inline Yurii_Telishevskyi/todo/src/app/todolists/todolists.component.ts]: Create string constant for this

**Review by KolosUK — COMMENTED** [review]
Delete all spec.ts files from your project

**Comment by KolosUK** [comment]
You can revert your 'Test commits' and don't push them, So revert all your test commits and keep your branch clearer.
Remember also for name your commits from previous lessons.

File: Yurii_Telishevskyi/todo/src/app/app.component.ts [hunk 9]
```diff
@@ -0,0 +1,14 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'todolist',
+  templateUrl: './app.component.html',
+  styleUrls: ['./app.component.scss']
+
```
Comment (NazarLuchyt) [inline Yurii_Telishevskyi/todo/src/app/app.component.ts]: Extra line. Try to write code without extra lines/whitespaces (read the documentation about clean code)

File: Yurii_Telishevskyi/todo/src/app/todocomponent/todocomponent.component.ts [hunk 10]
```diff
@@ -0,0 +1,34 @@
+import { Component, Input, Output, EventEmitter } from '@angular/core';
+
+
+@Component({
+  selector: 'app-todocomponent',
+  templateUrl: './todocomponent.component.html',
+  styleUrls: ['./todocomponent.component.scss']
+})
+export class TodocomponentComponent {
```
Comment (NazarLuchyt) [inline Yurii_Telishevskyi/todo/src/app/todocomponent/todocomponent.component.ts]: You don't need to use in the name word "component". The component is an entity in Angular

File: Yurii_Telishevskyi/todo/src/app/todolists/todolists.component.ts [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+export class TodolistsComponent {
+
+  title = 'todo';
+  todo: string;
+  text: string;
+  list: string[] = [];
+  btnAdd = 'Add';
+  i: number;
+  addToList() {
+    if (this.btnAdd === 'Add') {
```
Comment (NazarLuchyt) [inline Yurii_Telishevskyi/todo/src/app/todolists/todolists.component.ts]: Try to use another way to check this

File: Yurii_Telishevskyi/todo/src/app/todolists/todolists.component.ts [hunk 12]
```diff
@@ -0,0 +1,45 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'app-todolists',
+  templateUrl: './todolists.component.html',
+  styleUrls: ['./todolists.component.scss']
+})
+export class TodolistsComponent {
```
Comment (Magfiozy) [inline Yurii_Telishevskyi/todo/src/app/todolists/todolists.component.ts]: Why you use multiply? Do you have a lot of lists?

File: Yurii_Telishevskyi/todo/src/app/todocomponent/todocomponent.component.html [hunk 13]
```diff
@@ -1,6 +1,6 @@
-<div class="design1">
+<div class="flinch">
     <input type="checkbox"  (change)="checked()">
-    <span [class.line]="isGood">{{ items }}</span>
-    <button [disabled]="isGood" (click)="isEdited()" class = "m-1 p-1 btn btn-outline-primary"> Edit </button>
-    <button [disabled]="isGood" (click)="isDeleted()" class = "m-1 p-1 btn btn-outline-primary"> Delete </button>
+    <span [class.line]="onGood">{{ items }}</span>
```
Comment (NazarLuchyt) [inline Yurii_Telishevskyi/todo/src/app/todocomponent/todocomponent.component.html]: onGood is a boolean variable. Please use correct name for this.

File: Yurii_Telishevskyi/todo/src/app/todocomponent/todocomponent.component.html [hunk 14]
```diff
@@ -1,6 +1,6 @@
...
+    <button [disabled]="onGood" (click)="isEdited()" class = "m-1 p-1 btn btn-outline-primary"> Edit </button>
```
Comment (NazarLuchyt) [inline Yurii_Telishevskyi/todo/src/app/todocomponent/todocomponent.component.html]: isEdited - є редагованою. What should do this function? return boolean at least.

File: Yurii_Telishevskyi/todo/src/app/todocomponent/todocomponent.component.ts [hunk 15]
```diff
@@ -1,34 +1,32 @@
 import { Component, Input, Output, EventEmitter } from '@angular/core';
 
-
 @Component({
   selector: 'app-todocomponent',
   templateUrl: './todocomponent.component.html',
   styleUrls: ['./todocomponent.component.scss']
 })
 export class TodocomponentComponent {
   @Input() items: string;
```
Comment (NazarLuchyt) [inline Yurii_Telishevskyi/todo/src/app/todocomponent/todocomponent.component.ts]: items is an array? or a single value?

**Review by sidgar21 — DISMISSED** [review]
All is okey

**Review by sidgar21 — DISMISSED** [review]
Approve)

**Review by LTania — DISMISSED** [review]

**Review by taraswithjazz — DISMISSED** [review]

**Review by kblueberry — DISMISSED** [review]

File: Yurii_Telishevskyi/todo/src/app/app-routing.module.ts [hunk 16]
```diff
@@ -0,0 +1,11 @@
+import { NgModule } from '@angular/core';
+import { Routes, RouterModule } from '@angular/router';
+
+
+const routes: Routes = [];
+
```
Comment (KolosUK) [inline Yurii_Telishevskyi/todo/src/app/app-routing.module.ts]: Why u add module but don't use it.

File: Yurii_Telishevskyi/todo/src/app/todocomponent/todocomponent.component.ts (hunk 10 above)
Comment (KolosUK) [reply Yurii_Telishevskyi/todo/src/app/todocomponent/todocomponent.component.ts]: Not fixed

**Review by LTania — APPROVED** [review]

**Review by kblueberry — APPROVED** [review]

**Review by taraswithjazz — APPROVED** [review]

**Review by KolosUK — APPROVED** [review]

**Review by sidgar21 — APPROVED** [review]

**Review by mefissto — APPROVED** [review]

**Pull request merged**